lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorTargetSensitivity.cxx

```diff
@@ -0,0 +1,197 @@
+//                                               -*- C++ -*-
+/**
+ * @brief HSICEstimatorTargetSensitivity implements the target sensitivity HSIC indices
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/HSICEstimatorTargetSensitivity.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+CLASSNAMEINIT(HSICEstimatorTargetSensitivity)
+
+/* Default */
+HSICEstimatorTargetSensitivity::HSICEstimatorTargetSensitivity()
+  : HSICEstimatorImplementation()
+  , isAlreadyComputedPValuesAsymptotic_(false)
+{
+  // Nothing to do
+}
+
+/* Constructor */
+HSICEstimatorTargetSensitivity::HSICEstimatorTargetSensitivity(
+  const CovarianceModelCollection & covarianceList
+  , const Sample & X
+  , const Sample & Y
+  , const HSICStat & estimatorType
+  , const Function & filterFunction )
+  : HSICEstimatorImplementation(covarianceList, X, Y, estimatorType)
+  , isAlreadyComputedPValuesAsymptotic_(false)
+{
+  filterFunction_ =  filterFunction;
+  /* apply filter */
+  outputSample_ = filterFunction_(outputSample_);
+}
+
+/* Virtual constructor */
+HSICEstimatorTargetSensitivity* HSICEstimatorTargetSensitivity::clone() const
+{
+  return new HSICEstimatorTargetSensitivity(*this);
+}
+
+/* Compute the weight matrix from a sample */
+SquareMatrix HSICEstimatorTargetSensitivity::computeWeightMatrix(const Sample&) const
+{
+  /* Identity matrix */
+  const IdentityMatrix mat(n_);
+  return mat;
+}
+
+/* Compute the asymptotic p-values */
+void HSICEstimatorTargetSensitivity::computePValuesAsymptotic() const
```

Comment:
This is the *exact same code* as in the HSICEstimatorGlobalSensitivity::computePValueAsymptotic() method. No way to factorize it?

Reply:
A of factorizing this method would be to move it to the `HSICEstimatorImplementation` parent class, and override it in the `HSICEStimatorConditionalSensitivity` derived class. The two options were discussed several times, but no consensus was ever reached. Personally I don't have a specific preference.

Reply:
I tend to agree with Régis on this. I've moved the method to HSICEstimatorImplementation and overloaded it in HSICEstimatorConditionalSensitibity to throw (although the latter is not strictly necessary as no HSICEstimatorConditionalSensitivity method can call it).